sound/soc/sof/sof-acpi-dev.c

```diff
@@ -149,22 +133,18 @@ static const struct sof_ops_table mach_ops[] = {
 	{&sof_acpi_broadwell_desc, &sof_bdw_ops},
 #endif
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_BAYTRAIL)
-	{&sof_acpi_baytrail_desc, &sof_byt_ops, mfld_new_mach_data},
-	{&sof_acpi_baytrailcr_desc, &sof_byt_ops, mfld_new_mach_data},
-	{&sof_acpi_cherrytrail_desc, &sof_cht_ops, mfld_new_mach_data},
+	{&sof_acpi_baytrail_desc, &sof_byt_ops},
```

Comment:
better to remove *new_data from sof_ops_table thoroughly.

Reply:
ah yes good point. Thanks!